qt/applications/workbench/workbench/plotting/figureerrorsmanager.py

```diff
@@ -0,0 +1,130 @@
+        Relevant source, as of 10 July 2019:
+        https://github.com/matplotlib/matplotlib/blob/154922992722db37a9d9c8680682ccc4acf37f8c/lib/matplotlib/legend.py#L1201
+
+        :param parent_menu: The menu to which the actions will be added
+        :type parent_menu: QMenu
+        """
+        ax = self.canvas.figure.axes[0]  # type: matplotlib.axes.Axes
+        # if the ax is not a MantidAxes, and there are no errors plotted,
+        # then do not add any options for the menu
+        if not isinstance(ax, MantidAxes) and len(ax.containers) == 0:
+            return
+
+        error_bars_menu = QMenu(self.ERROR_BARS_MENU_TEXT, parent_menu)
+        error_bars_menu.addAction(self.SHOW_ERROR_BARS_BUTTON_TEXT,
+                                  partial(self._update_plot_after, self._toggle_all_errors, ax, make_visible=True))
+        error_bars_menu.addAction(self.HIDE_ERROR_BARS_BUTTON_TEXT,
+                                  partial(self._update_plot_after, self._toggle_all_errors, ax, make_visible=False))
+        parent_menu.addMenu(error_bars_menu)
+
+        self.active_lines = CurvesTabWidgetPresenter.get_active_lines(
+            ax, CurvesTabWidgetPresenter.get_errorbars_from_ax(ax))
+
+        # if there's more than one line plotted, then
+        # add a sub menu, containing an action to hide the
+        # error bar for each line
+        error_bars_menu.addSeparator()
+        add_later = []
+        for index, line in enumerate(self.active_lines):
+            if curve_has_errors(line):
+                curve_props = CurveProperties.from_curve(line)
+                # Add lines without errors first, lines with errors are appended later. Read docstring for more info
+                if not isinstance(line, ErrorbarContainer):
+                    action = error_bars_menu.addAction(line.get_label(), partial(
+                        self._update_plot_after, self.toggle_error_bars_for, ax, line))
+                    action.setCheckable(True)
+                    action.setChecked(not curve_props.hide_errors)
+                else:
+                    add_later.append((line.get_label(), partial(
+                        self._update_plot_after, self.toggle_error_bars_for, ax, line),
+                                      not curve_props.hide_errors))
+
+        for label, function, visible in add_later:
+            action = error_bars_menu.addAction(label, function)
+            action.setCheckable(True)
+            action.setChecked(visible)
+
+    def _toggle_all_errors(self, ax, make_visible):
+        for line in self.active_lines:
+            if curve_has_errors(line):
+                self.toggle_error_bars_for(ax, line, make_visible)
+
+    @staticmethod
+    def toggle_error_bars_for(ax, curve, make_visible=None):
+        curve_props = CurveProperties.from_curve(curve)
+        plot_kwargs = {key: value for key, value in curve_props.items() if key not in ['hide', 'hide_errors']}
```

Comment:
I think because we're not converting the `hide` argument into the `visible` Matplotlib kwarg if you set a curve to hidden in the plot options, and then toggle errorbars, the curve will no longer be hidden. I don't know if this is desired behaviour or not, but if we do want to keep the curve hidden in this case, we just need this line to be changed to
```python
plot_kwargs = curve_props.get_plot_kwargs()
```

Reply:
I am not sure about the behaviour, but this lets me reuse the same code as the presenter, which I'm all for